CountingBloomFilter.hpp

```diff
@@ -271,101 +249,122 @@ template<typename T>
 void
 CountingBloomFilter<T>::readFilter(const std::string& path)
 {
-	FILE* file;
-	if ((file = fopen(path.c_str(), "rb")) == nullptr) {
+	std::ifstream file(path);
+	if (!file) {
 		std::cerr << "ERROR: Failed to open file: " << path << "\n";
 		exit(EXIT_FAILURE);
 	}
 	readHeader(file);
-	struct stat buf;
-	if (fstat(fileno(file), &buf) != 0) {
-		std::cerr << "ERROR: Failed to open file: " << path << "\n";
-		exit(EXIT_FAILURE);
-	}
-	size_t arraySizeOnDisk = buf.st_size - sizeof(struct FileHeader);
-	if (arraySizeOnDisk != m_sizeInBytes) {
-		std::cerr << "ERROR: File size of " << path << " (" << arraySizeOnDisk << " bytes), "
-		          << "does not match size read from its header (" << m_sizeInBytes << " bytes).\n";
-		exit(EXIT_FAILURE);
-	}
-
-	size_t nread = fread(m_filter, arraySizeOnDisk, 1, file);
-	if (nread != 1 && fclose(file) != 0) {
+	m_filter.resize(m_sizeInBytes);
+	file.read(reinterpret_cast<char*>(m_filter.data()), m_sizeInBytes);
```

Comment:
```suggestion
	file.read(static_cast<char*>(m_filter.data()), m_sizeInBytes);
```
Does `static_cast` work here? Prefer `static_cast` over `reinterpret_cast` when possible.

Reply:
There is an error if i try to do that.

```../lib/bloomfilter/CountingBloomFilter.hpp:259:11: error: invalid static_cast from type 'unsigned char*' to type 'char*'
  file.read(static_cast<char*>(m_filter.data()), m_sizeInBytes);
```

Reply:
Sounds like `reinterpret_cast` is required.